### ISSUE TITLE
Oxlint reports incorrect errors when it receives a relative path starting with `..`

### ISSUE THREAD

**Issue by ah-yu** [oxc-project/oxc#1887]
```bash
test_repo % npx oxlint@latest --import-plugin ../test_repo 
× Failed to open file "test_repo/foo/bar/index.js" with error "No such file or directory (os error 2)"
help: Failed to open file "test_repo/foo/bar/index.js" with error "No such file or directory (os error 2)"
```
The error reported by `oxlint` appears to be incorrect. In fact, the file `test_repo/foo/bar/index.js` exists. It seems to be an issue related to `oxc_resolver`.

```js
// enhanced resolver
resolve.sync("../test_repo/foo/bar", "../main") // returns "../test_repo/foo/main.js"

// oxc_resolver
resolver.resolve("../test_repo/foo/bar", "../main").unwrap() // returns "test_repo/foo/main.js"
```

**Comment by IWANABETHATGUY** [comment]
https://github.com/oxc-project/oxc_resolver/blob/4585b370614fdfc8cd8c473e94c0aa35770f82ef/src/cache.rs#L192-L192, `fs.canonicalize` will strip the `../`

**Comment by IWANABETHATGUY** [comment]
the implementation of `oxc_resolver` does not looks wrong, call site could  `canonicalize` the path before resolving, here is  a example https://github.com/oxc-project/oxc_resolver/blob/4585b370614fdfc8cd8c473e94c0aa35770f82ef/examples/resolver.rs#L10-L10.

**Comment by IWANABETHATGUY** [comment]
I can replicate the issue with a relative path, could you provide a reproduce repo?

**Comment by ah-yu** [comment]
> could you provide a reproduce repo?

https://github.com/ah-yu/resolver-test

**Comment by Boshen** [comment]
I decided to disallow supplying paths pointing to parent directories, the intention is not clear and it will introduce undefined behaviour.
